code/datums/components/atmospheric_scanner.dm

```diff
@@ -0,0 +1,58 @@
+///Atoms with this component can signal it in order to get a scan of surrounding atmospherics sent to a specific mob.
+/datum/component/atmospheric_scanner
+	/// Controls if the analyzer requires being able to see it in order to obtain the results. The value is set in AddComponent.
+	var/visible = TRUE
```

Comment:
Visible is a little bit unclear what it wants. Maybe, `requires_sight` or something instead?

Reply:
requires_sight sounds good.